development/tools/pkg/dnscollector/wrappers.go

```diff
@@ -0,0 +1,101 @@
+package dnscollector
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	compute "google.golang.org/api/compute/v1"
+	dns "google.golang.org/api/dns/v1"
+)
+
+//ComputeServiceWrapper A wrapper for compute API service connections.
+type ComputeServiceWrapper struct {
+	Context context.Context
+	Compute *compute.Service
+}
+
+//DNSServiceWrapper A wrapper for dns API service connections.
+type DNSServiceWrapper struct {
+	Context context.Context
+	DNS     *dns.Service
+}
+
+func (csw *ComputeServiceWrapper) lookupRegions(project, pattern string) ([]string, error) {
+	call := csw.Compute.Regions.List(project)
+	if pattern != "" {
+		call = call.Filter("name: " + pattern)
+	}
+
+	var regions []string
+	f := func(page *compute.RegionList) error {
+		for _, v := range page.Items {
+			regions = append(regions, v.Name)
+		}
+		return nil
+	}
+
+	if err := call.Pages(csw.Context, f); err != nil {
+		return nil, err
+	}
+	return regions, nil
+}
+
+func (csw *ComputeServiceWrapper) lookupIPAddresses(project string, region string) ([]*compute.Address, error) {
```

Comment:
Same about callback here.

Reply:
After a discussion a decision is to stay with a list of address objects.